dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1379,6 +1379,14 @@ public Workflow manualRegister(@ApiParam(hidden = true) @Auth User user,
                         }
 
                     }
+
+                    //store dag and tool table json
+                    if (existingTag.getDagJson() == null) {
+                        existingTag.setDagJson(getWorkflowDag(Optional.of(user), w.getId(), existingTag.getId()));
```

Comment:
1. It feels weird to me to call a REST endpoint like this when you're already in a different REST endpoint. It's ultimately just a function being called so maybe it's OK, but, for example, depending on how the UnitOfWork annotation is implemented, you might be ending up with a nested transaction. There are other things that you can also inject into a method, like `@Context`, which we aren't doing now, but could someday, and that might not work.
2. That method already caches for you, so although it's harmless, you're writing to the column twice.

You already have the toolsJSONTable result on line 1364; you don't need to calculate it again. The LanguageHandlerInterface there can be used to to get the DAG.